c++/src/capnp/rpc.c++

```diff
@@ -3147,6 +3206,19 @@ private:
       // that we already removed it and re-allocated the ID to something else. So, we should ignore
       // the `Return`. But we might want to make note to stop using these hints, to protect against
       // the (again, remote) possibility of our ID space wrapping around and leading to confusion.
+      if (questions.isLow(questionId)) {
+        // The low bit of the questionId is set, meaning that it is a realtime message. In this case,
+        // we must send a Finish message.
```

Comment:
You should check if `noFinishNeeded` is set on the return message. If so, you don't need to send a Finish! The `noFinishNeeded` bit is commonly set when a return message contains no capabilities, which should always be the case for streaming calls since they don't return anything at all.